rules/linux/lnx_network_service_scanning.yml

```diff
@@ -0,0 +1,31 @@
+title: Linux Network Service Scanning
+id: 3e102cd9-a70d-4a7a-9508-403963092f31
+status: experimental
+description: Detects enumeration of local or remote network services.
+author: Alejandro Ortuno, oscd.community
+date: 2020/10/21
+references:
+  - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1046/T1046.md
+logsource:
+  category: process_creation
+  product: linux
+detection:
+  selection_1:
+    ProcessName|endswith:
+      - '/cat'
+  selection_2:
+    ProcessName|endswith:
+      - '/nmap'
+  selection_3:
+    ProcessName|endswith:
+      - '/telnet'
+  selection_4:
+    ProcessName|endswith:
+      - '/nc'
```

Comment:
that one gonna be extremely noisy as well. I would add filter for listening flag, which is `-l` or `--listen`. could be:

```
selection_4:
    ProcessName|endswith: '/nc'
filter:
    CommandLine|contains: 'l'
```

> (and add it to condition with `and no filter` for selection_4)

no spaces, no dashes, because that could be `-nlvp` or `-nvlp` or any other combinations.

At the same time, it's gonna generate False Negatives in case an adversary will scan a resource using a domain name that includes `l` character. How would you modify the rule to protect it against such cases?
Or maybe there is a better solution to solve the issue? Let's try to figure that out! (:

Reply:
@yugoslavskiy I think on this case will make more sense if we move into the syscalls and specifically into the `connect` one https://linux.die.net/man/2/connect? In that case we will move out from specific command lines but into the actual connect event.

If we look for example at the `socket_events` table on oquery: Linux socket auditing section https://osquery.readthedocs.io/en/stable/deployment/process-auditing/ we will have all relevant details in there.

Would not look at the binding one (the listening flag you were mention) as this technique is for remote scanning? Let me know what you think before tackling the refactor of the rules.

Reply:
@alejandroortuno that totally makes sense! 

> Would not look at the binding one (the listening flag you were mention) as this technique is for remote scanning?

Yes, `connect` is used for scanning, while `bind` is for listening on some port (that is used by `-l` flag). So that's our option.

I think that it's not a good time to use osquery, but there is a few examples of rules for syscalls detection that rely on auditd. We can refer to a [specific audit policy](https://github.com/Neo23x0/auditd/blob/master/audit.rules#L182-L183) in the definition of the logsource and then use it in the rule:

```
logsource:
    product: linux
    service: auditd
   definition: 'configure these rules https://github.com/Neo23x0/auditd/blob/master/audit.rules#L182-L183'
detection:
    selection:
        type: 'SYSCALL'
        exe|endswith:
            - '/nc'
            - '/netcat'
        key: 'network_connect_4'
```

At the same time, if you would like to use both `process_creation` and `auditd` logsources, you will need to use [Rule Collections](https://github.com/Neo23x0/sigma/wiki/Specification#rule-collections) to make it work. You can find a few examples of this kind of rules by pattern `action: global` (: